themes/default/content/blog/troubleshooting-builds-with-diagnostics/index.md

```diff
@@ -0,0 +1,154 @@
+---
+title: "Troubleshooting Builds With Diagnostics"
+date: 2022-03-29T12:00:55-06:00
+draft: false
+meta_desc: Explore how to gather diagnostic data to debug your Pulumi programs.
+meta_image: meta.png
+authors:
+    - laura-santamaria
+tags:
+    - troubleshooting
+---
+
+One of the most common things a developer, an operations professional, a security researcher, and a quality assurance engineer all have in common is the pain of troubleshooting. No matter where you are in an IT organization, you probably have cursed your system at some point as you stare at a wall of characters representing an error somewhere. Whenever humans interact with a system, we introduce chaos, and the more complex the system, the more likely the system's diagnostic data is complex and multi-layered. Pulumi, as a system that represents some of the more complex systems we can make, is no exception. So let's demystify diagnostics in Pulumi.
+
+<!--more-->
+
+## Getting data
+
+First, if you don't know how to get diagnostic data from a Pulumi run, our [troubleshooting page]({{< relref "/docs/troubleshooting" >}}) is a good place to start. There's a section specifically on [diagnostics]({{< relref "/docs/troubleshooting#diagnosing-issues" >}}). If you're in a hurry, though, here's the fast answer that gives you every single piece of data you could possibly want:
+
+```bash
+TF_LOG=TRACE PULUMI_DEBUG_PROMISE_LEAKS=true pulumi <command> -v=11 --logflow --logtostderr 2>&1 | tee -a pulumi_log.txt
```

Comment:
I’m not sure that in practice this is ever what a user will want to do. The command in the docs is the most practically useful command to run. I get that it’s being presented here as “the kitchen sink” - but I think the practical experience of using this will be bad enough that I’m not sure it helps to lead with this (vs just explain the options for getting additional logs).

Reply:
Hm, David had thought adding this in would be helpful as some folks are just looking for a one-liner, and that made sense (though I think people generally head to the docs for that). But I see your point. I'll trim it down to a more tame version of the command as an in-between.